WalletWasabi.Gui/Controls/WalletExplorer/CoinListView.xaml

```diff
@@ -74,31 +74,19 @@
       </StackPanel>
       <controls:BusyIndicator IsBusy="{Binding IsCoinListLoading}" Text="Loading...">
         <controls:ExtendedListBox Items="{Binding Coins}" VirtualizationMode="None" SelectedItem="{Binding Path=SelectedCoin, Mode=TwoWay}">
-          <controls:ExtendedListBox.ContextMenu>
-            <ContextMenu>
-              <!--Enqueuing needs password, TODO: jump to password box OR display pw box in context menu + send button-->
-              <MenuItem Command="{Binding EnqueueCoin}" IsVisible="false">
-                <MenuItem.Header>
-                  <Grid>
-                    <Grid.ColumnDefinitions>
-                      <ColumnDefinition />
-                      <ColumnDefinition />
-                    </Grid.ColumnDefinitions>
-                    <TextBlock Text="Enqueue Coin, password:" VerticalAlignment="Center" />
-                    <TextBox Width="50" Grid.Column="1" />
-                  </Grid>
-                </MenuItem.Header>
-              </MenuItem>
-              <MenuItem Header="Dequeue from CoinJoin" Command="{Binding DequeueCoin}">
-                <MenuItem.Icon>
-                  <DrawingPresenter HorizontalAlignment="Left" Height="16" Width="16" Stretch="Fill" Drawing="{StaticResource CoinListView_DequeueCoin}" />
-                </MenuItem.Icon>
-              </MenuItem>
-            </ContextMenu>
-          </controls:ExtendedListBox.ContextMenu>
           <controls:ExtendedListBox.ItemTemplate>
             <DataTemplate>
               <Grid>
+                <Grid.ContextMenu>
+                  <!-- If you want to add more items to the context menu and you want the Send tab to show it too, then place IsVisible="{Binding InCoinJoinContainer}" directly to the Dequeue MenuItem. -->
+                  <ContextMenu IsVisible="{Binding InCoinJoinContainer}">
```

Comment:
I'm afraid that this could trigger the same bug still...

Reply:
Yes, I noted it in my PR description. However it would be better to test it if a new solution is needed or not, since in your PR the context menu is visible and thus when you try to dequeue it from the Send tab you get this:

![image](https://user-images.githubusercontent.com/9156103/70786050-b83da800-1dc6-11ea-9452-1a1cd8e23416.png)

So we aren't getting away with this :)

Reply:
I wonder if IsEnable could work instead?

Reply:
No, I tried that first.

Reply:
Oh alright... no getting around it indeed :+1:

Reply:
As I noted in the comment we can apply `IsVisible` for the `MenuItem` itself, just like we do it in the receive tab to test if hardware wallet options should be fired up or not. However that results in a small rectangle in the screen, which is not optimal, but better than freezing.

Reply:
I was first trying to make sure we can dequeue from the send tab, too, but that's quite difficult, so I decided to not abandon that path, but if needed, risking some more clutter it could be attempted.

However the clean solution would be to not show up the context menu when the coin list is from the send tab.

Another path could be to figure out some new functionality that justifies having the context menu there, eg, the toggle expander like in the receive tab and this way we can disable it in item menu level.

Reply:
I went with solution 3. That makes everyone happy and avoids the possible OSX deadlock: https://github.com/zkSNACKs/WalletWasabi/pull/2849#issuecomment-565370496